### PR TITLE
support alignment ops when processing statements in `RegisterUse`

### PR DESCRIPTION
**Pull request by Ptival**
For some reason we were very conservative in our support or abstract
operations over the processor state in the `RegisterUse` analysis.

In particular, we were failing to process code such as:

r23 := (bv_and r21 (0xfffffffffffffff0 :: [64]))

whose goal is to align the value in r21 at a 16-byte boundary.

This resulted in us failing to analyze some code that was realigning its
stack pointer.  With this change, the same code succeeds at propagating
the abstract stack pointer offset forward.

### CODE REVIEW

File: base/src/Data/Macaw/Analysis/RegisterUse.hs [hunk 1]
```diff
@@ -678,6 +679,8 @@ processApp aid app = do
       setAssignVal aid (FrameExpr (o+fromInteger c))
     BVSub _ (FrameExpr o) (IVCValue (BVCValue _ c)) ->
       setAssignVal aid (FrameExpr (o-fromInteger c))
+    BVAnd _ (FrameExpr o) (IVCValue (BVCValue _ c)) ->
+      setAssignVal aid (FrameExpr (o Bits..&. memInt (fromInteger c)))
```
Comment (RyanGlScott) [inline base/src/Data/Macaw/Analysis/RegisterUse.hs:683]: For consistency with the code above, you could also write this as:

```suggestion
      setAssignVal aid (FrameExpr (o Bits..&. fromInteger c))
```

File: base/src/Data/Macaw/Analysis/RegisterUse.hs [hunk 2]
```diff
@@ -678,6 +679,8 @@ processApp aid app = do
       setAssignVal aid (FrameExpr (o+fromInteger c))
     BVSub _ (FrameExpr o) (IVCValue (BVCValue _ c)) ->
       setAssignVal aid (FrameExpr (o-fromInteger c))
+    BVAnd _ (FrameExpr o) (IVCValue (BVCValue _ c)) ->
```
Comment (RyanGlScott) [inline base/src/Data/Macaw/Analysis/RegisterUse.hs:682]: Does it also make sense to add a `BVAnd _ (IVCValue (BVCValue _ c)) (FrameExpr o)` case for symmetry with the two `BVAdd` cases above? (I'm not sure if this case would ever occur in practice, but I did find myself wondering about its omission here.)

**Review by RyanGlScott — APPROVED** [review]
This is definitely an improvement.

File: base/src/Data/Macaw/Analysis/RegisterUse.hs (hunk 1 above)
Comment (Ptival) [reply base/src/Data/Macaw/Analysis/RegisterUse.hs:683]: Agreed!

File: base/src/Data/Macaw/Analysis/RegisterUse.hs (hunk 2 above)
Comment (Ptival) [reply base/src/Data/Macaw/Analysis/RegisterUse.hs:682]: I agree that in practice this version is the most likely to be generated by Macaw, but it doesn't hurt to add, so I will.

**Comment by Ptival** [comment]
@RyanGlScott I have addressed your comments, and also added two comments that I fell will help me if I ever need to look at this code again.

**Pull request merged**